packages/tx/src/eip1559Transaction.ts

```diff
@@ -228,6 +230,9 @@ export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMark
 
     const freeze = opts?.freeze ?? true
     if (freeze) {
+      if (this.isSigned()) {
+        this._cachedHash = this.hash()
+      }
```

Comment:
Hi there, thanks for the PR! 😄 

The cache creation for the hash shouldn't be done in the constructor but in the `hash()` function itself on the first execution (same for `Block`). Otherwise we are actually adding *overhead* for use cases where the`hash()` function is not needed.

If your problem is that the object actually will get frozen after the constructor is called:

I did some experiments on this and I guess it should work to wrap the `_cachedHash` property into some object, which will *then* still be modifiable. See the following results:

```shell
> const inner = { hash: '123' }
'use strict'
> const outer = { inner, test: 5 }
undefined
> outer.test = 4
4
> outer
{ inner: { hash: '123' }, test: 4 }
> Object.freeze(outer)
{ inner: { hash: '123' }, test: 4 }
> outer.test = 3
3
> outer
{ inner: { hash: '123' }, test: 4 }
> outer.inner.hash = '345'
'345'
> outer
{ inner: { hash: '345' }, test: 4 }
```

So as one can see the `inner` object `hash` property can still be modified on a frozen object (maybe someone else should nevertheless have an additional look here to see if this has no side effects whatsoever. Can't think of any thow atm).

Maybe choose some structure like:

```typescript
private cache = { hash: undefined }
```

Then we can eventually expand this with other stuff. 😄

Reply:
Sounds good - will give it a try

Reply:
@holgerd77 I have an implementation in `packages/block/src/header.ts` based on your suggestion - tests are passing and I think it makes sense. Wanted to get your feedback on the approach, and if it looks good, I'll add it in `tx` and add the documentation comments. Thanks!

Reply:
Yes, that looks good, ready to proceed! 😄